### PR TITLE
[backbone] add hashChange to HistoryOptions

### PR DESCRIPTION
**Pull request by MarcCelani-at**
This adds in the hashChange option to HistoryOptions, which is used by Backbone to here: https://github.com/jashkenas/backbone/blob/master/backbone.js#L1839

### CODE REVIEW

**Comment by khasanovbi** [comment]
LGTM, it would be cool if you add some simple test for this

**Review by khasanovbi — APPROVED** [review]

**Comment by jgonggrijp** [comment]
Relevant documentation here: https://backbonejs.org/#History-start

**Comment by MarcCelani-at** [comment]
to be honest, I don't personally use backbone so idk how to write a good test :P I just noticed a @ts-expect-error in our code because this was missing so decided to add it. From my side its not urgent so if it takes some [reasonable] amount of time to get approved thats fine. Otherwise, I'll have to dig in to write a test or maybe one of the maintainers can help get this across the finish line? Anyways, thanks!

**Comment by jgonggrijp** [comment]
@MarcCelani-at I think it will suffice to just write a couple of lines where you pass a boolean `hashChange` option to `Backbone.history.start` (see linked documentation in my previous comment). After all, we're only verifying that the types check. Logic is already tested in Backbone proper. You can take example from pre-existing tests that look like standalone expressions.

**Review by khasanovbi — APPROVED** [review]

**Comment by MarcCelani-at** [comment]
I'll admit I was too lazy to figure out how to run the tests locally, so waiting on CI here. Sorry folks.

**Comment by MarcCelani-at** [comment]
Ready to merge

**Pull request merged**
